src/EntityComponentManager_TEST.cc

```diff
@@ -2204,6 +2204,72 @@ TEST_P(EntityComponentManagerFixture, Descendants)
   }
 }
 
+//////////////////////////////////////////////////
+TEST_P(EntityComponentManagerFixture,
+       IGN_UTILS_TEST_DISABLED_ON_WIN32(UpdatePeriodicChangeCache))
+{
+  Entity e1 = manager.CreateEntity();
+  auto c1 = manager.CreateComponent<IntComponent>(e1, IntComponent(123));
+
+  std::unordered_map<ComponentTypeId,
+                        std::unordered_set<Entity>> changeTracker;
+
+  // No periodic changes keep cache empty.
+  manager.UpdatePeriodicChangeCache(changeTracker);
+  EXPECT_EQ(changeTracker.size(), 0);
```

Comment:
This is causing a compiler warning
```
In file included from /home/jenkins/workspace/ignition_gazebo-ci-pr_any-ubuntu_auto-amd64/ign-gazebo/src/EntityComponentManager_TEST.cc:18:
/home/jenkins/workspace/ignition_gazebo-ci-pr_any-ubuntu_auto-amd64/ign-gazebo/test/gtest/include/gtest/gtest.h: In instantiation of ‘testing::AssertionResult testing::internal::CmpHelperEQ(const char*, const char*, const T1&, const T2&) [with T1 = long unsigned int; T2 = int]’:
/home/jenkins/workspace/ignition_gazebo-ci-pr_any-ubuntu_auto-amd64/ign-gazebo/test/gtest/include/gtest/gtest.h:1554:23:   required from ‘static testing::AssertionResult testing::internal::EqHelper::Compare(const char*, const char*, const T1&, const T2&) [with T1 = long unsigned int; T2 = int; typename std::enable_if<((! std::is_integral<_Tp>::value) || (! std::is_pointer<_Dp>::value))>::type* <anonymous> = 0]’
/home/jenkins/workspace/ignition_gazebo-ci-pr_any-ubuntu_auto-amd64/ign-gazebo/src/EntityComponentManager_TEST.cc:2219:3:   required from here
/home/jenkins/workspace/ignition_gazebo-ci-pr_any-ubuntu_auto-amd64/ign-gazebo/test/gtest/include/gtest/gtest.h:1527:11: warning: comparison of integer expressions of different signedness: ‘const long unsigned int’ and ‘const int’ [-Wsign-compare]
 1527 |   if (lhs == rhs) {
      |       ~~~~^~~~~~
```

```suggestion
  EXPECT_EQ(changeTracker.size(), 0u);
```
Here and everywhere where the comparison is with an unsigned int.

Reply:
[a6e85ba](https://github.com/gazebosim/gz-sim/pull/2001/commits/a6e85ba8e0032bf431129dd4ef9f86c25d528c49)